### PR TITLE
Brush

### PR DESCRIPTION
**Pull request by tingxuanz**
Hi @rowlandm this is a pull request from brush to zoom.

you can review how brush works.

### CODE REVIEW

File: css/main.css [hunk 1]
```diff
@@ -25,5 +25,19 @@ body {
 }
 
 .dot {
-  stroke: #000;
+  stroke: gray;
```
Comment (rowlandm) [inline css/main.css]: Probably don't want to make it gray to select. Make it normal. better UX. The user may not think that these dots are usable.

File: lib/index.js [hunk 2]
```diff
@@ -202,7 +204,63 @@ setup_zoom = function(graph){
...
+
+  var brush = d3.svg.brush()
+       .x(graph.scaleX)
+       .y(graph.scaleY)
+       .on("brushend", function() {
+         var x = graph.scaleX;
+         var y = graph.scaleY;
+
+         var extent = brush.extent(); //extent contains data of the brush area
+         x.domain([extent[0][0], extent[1][0]]);
```
Comment (rowlandm) [inline lib/index.js:230]: What do these do?  A one liner would be great for those not in the know.

File: lib/index.js [hunk 3]
```diff
@@ -202,7 +204,63 @@ setup_zoom = function(graph){
...
+         y.domain([extent[0][1], extent[1][1]]);
+
+         var xAxis = d3.svg.axis()
+                      .scale(x)
+                      .orient("bottom");
+         var yAxis = d3.svg.axis()
+                      .scale(y)
+                      .orient("left");
+
+         svg.transition().duration(500)
```
Comment (rowlandm) [inline lib/index.js]: Should set the duration as a variable and have it at the top of the function.

File: lib/index.js [hunk 4]
```diff
@@ -202,7 +204,63 @@ setup_zoom = function(graph){
...
+           .select(".xAxis")
+           .call(xAxis);
+
+         svg.transition().duration(500)
+           .select(".yAxis")
+           .call(yAxis);
+
+         svg.selectAll(".dot")
+            .transition()
+            .duration(500)
```
Comment (rowlandm) [inline lib/index.js]: Same here and above too.

File: lib/index.js [hunk 5]
```diff
@@ -202,7 +204,63 @@ setup_zoom = function(graph){
...
+            })
+            .attr("cy", function(d) {
+              var yDomain;
+              yDomain = d[options.yDomain];
+              return y(yDomain);
+            });
+
+            d3.event.target.clear();
+            d3.select(this).call(d3.event.target);
+            //d3.selectAll(".brush").remove();
```
Comment (rowlandm) [inline lib/index.js]: Please remove completely.

File: lib/index.js [hunk 6]
```diff
@@ -391,27 +449,50 @@ setup_zoom = function(graph){
...
 
+
   setup_graph = function(graph){
     //setup all the graph elements
     graph = setup_margins(graph);
     graph = setup_svg(graph);
     graph = setup_x_axis(graph);
     graph = setup_y_axis(graph);
-
+    //graph.toggleFlag = false;
```
Comment (rowlandm) [inline lib/index.js]: Please remove.

File: lib/index.js [hunk 7]
```diff
@@ -391,27 +449,50 @@ setup_zoom = function(graph){
...
     /*
     if you append your zoom/brush after your data points, the zoom/brush overlay will catch all of the pointer events,
     and your tooltips will disappear on hover.
     You want to append the zoom/brush before your data points, so that pointer events on the data generate a tooltip,
     and those on the overlay generate a zoom/brush.
     */
+    var svg = graph.svgForPCA;
+    svg.append("text")
+      .attr("y", 0)
```
Comment (rowlandm) [inline lib/index.js]: Do you want to have these as variables?

**Review by rowlandm — CHANGES_REQUESTED** [review]
7 things to have a look at. Most of them are minor.

File: css/main.css (hunk 1 above)
Comment (tingxuanz) [reply css/main.css]: Shall I setup the stroke as black? I think black is usually normal color for stroke.

File: lib/index.js (hunk 2 above)
Comment (tingxuanz) [reply lib/index.js:230]: I will add some comments in next commit

File: lib/index.js (hunk 3 above)
Comment (tingxuanz) [reply lib/index.js]: You mean the duration time (ie. 500) or the whole svg.transition().duration()? If it is the whole svg.transition().duration(), I don't think we can set it as a variable.

File: css/main.css (hunk 1 above)
Comment (rowlandm) [reply css/main.css]: That sounds fine.

File: lib/index.js (hunk 3 above)
Comment (rowlandm) [reply lib/index.js]: The 500 part only :)

**Review by rowlandm — APPROVED** [review]

**Pull request merged**